### PR TITLE
refactor: improve search hook handling

### PR DESCRIPTION
**Pull request by realSpok**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-customizer/src/decorators/actions/collection.ts [hunk 1]
```diff
@@ -65,13 +65,14 @@ export default class ActionCollectionDecorator extends CollectionDecorator {
...
 
     // Convert DynamicField to ActionField in successive steps.
-    let dynamicFields: DynamicField[];
+    let dynamicFields: DynamicField[] = action.form.map(c => ({ ...c }));
+
+    if (metas?.searchField) {
+      // in the case of a search hook,
+      // we don't want to rebuild all the fields. only the one searched
+      dynamicFields = dynamicFields.filter(field => field.label === metas.searchField);
+    }
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/actions/collection.ts:74]:
```suggestion
    let dynamicFields: DynamicField[] = action.form.map(c => ({ ...c }));

    if (metas?.searchField) {
      // in the case of a search hook,
      // we don't want to rebuild all the fields. only the one searched
      dynamicFields = dynamicFields.find(field => field.label === metas.searchField);
    }
```

Comment (realSpok) [reply packages/datasource-customizer/src/decorators/actions/collection.ts:74]: I did think about this but what if there are multiple fields with the same name: I know this should not be supported but there is nothing stoping someone to do so

**Review by Thenkei — APPROVED** [review]
👾

**Review by Thenkei — APPROVED** [review]
- ✅&nbsp;Code review - It eases readability. We argue about creating a dedicated handler, for now, it computes every fields.
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ❔&nbsp;PR linked to the clickup task

**Pull request merged**